packages/kit/src/runtime/server/page/load_data.js

```diff
@@ -186,6 +169,23 @@ export async function load_data({
 				}
 			});
 
+			// ensure that excluded headers can't be read
+			const get = response.headers.get;
```

Comment:
I'm wondering why the proxy is used instead of monkey patching the other methods as well - convenience?

Also: Should we apply this check in DEV only?

Reply:
It would be more convenient to monkey-patch the `Response` methods instead of using the proxy, but we get 'illegal invocation' errors. For whatever reason that doesn't apply to `Headers` so we may as well do the convenient thing there.

I think it's fine to keep the error in production, since it doesn't result in extra code shipped to the client and might help someone diagnose a bug